src/platform/plugins/shared/discover/public/plugin.tsx

```diff
@@ -270,7 +273,11 @@ export class DiscoverPlugin
     return {
       locator: this.locator,
       DiscoverContainer: (props: DiscoverContainerProps) => (
-        <DiscoverContainerInternal getDiscoverServices={getDiscoverServicesInternal} {...props} />
+        <DiscoverContainerInternal
+          getDiscoverServices={getDiscoverServicesInternal}
+          enableEsqlByDefault={this.enableEsqlByDefault}
```

Comment:
Not sure what is the best way to pass this prop to `get_state_defaults`. 
FYI @elastic/kibana-data-discovery